src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -2623,12 +2630,15 @@ void ApplicationManagerImpl::UnregisterApplication(
           logger_, "There is no more SDL4 apps with device handle: " << handle);
 
       RemoveAppsWaitingForRegistration(handle);
+      ClearCommandListAndSendOnAppUnregNotificationToHMI(
```

Comment:
@KVGrygoriev I really dont like functions with more than one responsibility.
I suggest you to update this code like:
```
if (!app) { 
  LOG4CXX_DEBUG( 
       logger_, "There is no more SDL4 apps with device handle: " << handle); 
       RemoveAppsWaitingForRegistration(handle); 
}

commands_holder_->Clear(app_to_remove); 
MessageHelper::SendOnAppUnregNotificationToHMI(app_to_remove, is_unexpected_disconnect, *this);  

if (!app) { 
	SendUpdateAppList(); 
}
```

Or even like this if `SendUpdateAppList` call order is not necessary for this case:
```
if (!app) { 
  LOG4CXX_DEBUG( 
       logger_, "There is no more SDL4 apps with device handle: " << handle); 
       RemoveAppsWaitingForRegistration(handle); 
       SendUpdateAppList(); 
}

commands_holder_->Clear(app_to_remove); 
MessageHelper::SendOnAppUnregNotificationToHMI(app_to_remove, is_unexpected_disconnect, *this);  
```